beacon-chain/db/kv/blobs_test.go

```diff
@@ -0,0 +1,101 @@
+	cfg.MinEpochsForBlobsSidecarsRequest = 2
+	params.OverrideBeaconNetworkConfig(cfg)
+	db := setupDB(t)
+
+	sidecars := make([]*ethpb.BlobsSidecar, 0)
+	numSlots := uint64(cfg.MinEpochsForBlobsSidecarsRequest) * uint64(params.BeaconConfig().SlotsPerEpoch)
+	for i := uint64(0); i < numSlots; i++ {
+		// There can be multiple blobs per slot with different block roots, so we create some
+		// in order to have a thorough test.
+		root1 := bytesutil.ToBytes32([]byte(fmt.Sprintf("foo-%d", i)))
+		root2 := bytesutil.ToBytes32([]byte(fmt.Sprintf("bar-%d", i)))
+		sidecars = append(sidecars, &ethpb.BlobsSidecar{
+			BeaconBlockRoot: root1[:],
+			BeaconBlockSlot: types.Slot(i),
+			Blobs:           make([]*enginev1.Blob, 0),
+			AggregatedProof: make([]byte, 48),
+		})
+		sidecars = append(sidecars, &ethpb.BlobsSidecar{
+			BeaconBlockRoot: root2[:],
+			BeaconBlockSlot: types.Slot(i),
+			Blobs:           make([]*enginev1.Blob, 0),
+			AggregatedProof: make([]byte, 48),
+		})
+	}
+	ctx := context.Background()
+	for _, blobSidecar := range sidecars {
+		require.NoError(t, db.SaveBlobsSidecar(ctx, blobSidecar))
+		require.Equal(t, true, db.HasBlobsSidecar(ctx, bytesutil.ToBytes32(blobSidecar.BeaconBlockRoot)))
+	}
+
+	// We check there are only two blob sidecars stored at slot 0, as an example.
+	keyPrefix := append(bytesutil.SlotToBytesBigEndian(0), bytesutil.SlotToBytesBigEndian(0)...)
+	numBlobs := countBlobsWithPrefix(t, db, keyPrefix)
+	require.Equal(t, 2, numBlobs)
+
+	// Attempting to save another blob sidecar with slot 0 and a new block root should result
+	// in three blob sidecars stored at slot 0. This means we are NOT overwriting old data.
+	root := bytesutil.ToBytes32([]byte("baz-0"))
+	sidecar := &ethpb.BlobsSidecar{
+		BeaconBlockRoot: root[:],
+		BeaconBlockSlot: types.Slot(0),
+		Blobs:           make([]*enginev1.Blob, 0),
+		AggregatedProof: make([]byte, 48),
+	}
+	require.NoError(t, db.SaveBlobsSidecar(ctx, sidecar))
+	require.Equal(t, true, db.HasBlobsSidecar(ctx, bytesutil.ToBytes32(sidecar.BeaconBlockRoot)))
+
+	numBlobs = countBlobsWithPrefix(t, db, keyPrefix)
+	require.Equal(t, 3, numBlobs)
+
+	// Now, we attempt to save a blob sidecar with slot = MAX_SLOTS_TO_PERSIST_BLOBS. This SHOULD cause us to
+	// overwrite ALL old data at slot 0, as slot % MAX_SLOTS_TO_PERSIST_BLOBS will be equal to 0.
+	// We should expect a single blob sidecar to exist at slot 0 after this operation.
+	root = bytesutil.ToBytes32([]byte(fmt.Sprintf("foo-%d", numSlots)))
+	sidecar = &ethpb.BlobsSidecar{
+		BeaconBlockRoot: root[:],
+		BeaconBlockSlot: types.Slot(numSlots),
+		Blobs:           make([]*enginev1.Blob, 0),
+		AggregatedProof: make([]byte, 48),
+	}
+	require.NoError(t, db.SaveBlobsSidecar(ctx, sidecar))
+	require.Equal(t, true, db.HasBlobsSidecar(ctx, bytesutil.ToBytes32(sidecar.BeaconBlockRoot)))
+
+	keyPrefix = append(bytesutil.SlotToBytesBigEndian(0), bytesutil.SlotToBytesBigEndian(64)...)
```

Comment:
You should not append `64` because the original sidecar at round buffer's `0` value had a different slot.

Reply:
In this example, the key we are expecting is 0 ++ 64 ++ bytes32(foo-64). A key contains the round buffer, the slot, and the block root. The new value we are storing at round buffer 0 will be under this new key, so this is the correct value to expect